be/src/exec/pipeline/driver_limiter.cpp

```diff
@@ -14,14 +14,21 @@
 
 #include "exec/pipeline/driver_limiter.h"
 
+#include "util/starrocks_metrics.h"
+
 namespace starrocks::pipeline {
 
+DriverLimiter::DriverLimiter(int max_num_drivers) : _max_num_drivers(max_num_drivers) {
+    REGISTER_GAUGE_STARROCKS_METRIC(pipe_drivers, [this]() { return num_total_drivers(); });
```

Comment:
DriverLimiter is not a singleton implementation, looks like not proper to repeatedly register the metric again and again for every DriverLimiter instance.

Reply:
`DriverLimiter` is only held by `ExecEnv`.

Reply:
And `REGISTER_GAUGE_STARROCKS_METRIC` will only register a new metric when it does not exist.

Reply:
Let me move `REGISTER_GAUGE_STARROCKS_METRIC` after initializing `_driver_limiter` in `Exec::Env`.